src/store.c

```diff
@@ -357,27 +356,8 @@ static int p11prov_store_load(void *pctx, OSSL_CALLBACK *object_cb,
     case CKO_PUBLIC_KEY:
     case CKO_PRIVATE_KEY:
         object_type = OSSL_OBJECT_PKEY;
-        type = p11prov_obj_get_key_type(obj);
```

Comment:
I would prefer to keep this code here and simply add a check with p11prov_obj_is_rsa_pss() in the CKK_RSA case.

Reply:
Good point. This is also an artifact from previous iteration as  this predates the time I had the `p11prov_obj_is_rsa_pss()` function. Will change.